Mathlib/Init.lean

```diff
@@ -0,0 +1,25 @@
+/-
+This is the root file in Mathlib: it is imported by virtually *all* Mathlib files.
+
+For this reason, the imports of this files are carefully curated.
+Any modification involving a change in the imports of this file should be discussed beforehand.
+
+Here are some general guidelines:
+* no bucket imports (e.g. `Batteries`/`Lean`/etc);
+* every import needs to have a comment explaining why the import is there;
+* strong preference for avoiding files that themselves have imports beyond `Lean`, and
+  any exception to this rule should by accompanied by a comment explaining the transitive imports.
+-/
+
+-- All these linters are meant to run on all of mathlib, are active by default,
+-- and have no imports beyond `Lean`.
+import Mathlib.Tactic.Linter.OldObtain
+import Mathlib.Tactic.Linter.RefineLinter
+-- xxx: this file imports Batteries.Lean.HashSet; is this acceptable?
+import Mathlib.Tactic.Linter.HashCommandLinter
+import Mathlib.Tactic.Linter.GlobalAttributeIn
+-- xxx: this file imports Batteries.Data.String.Matcher and Batteries.Tactic.Lint; acceptable?
+import Mathlib.Tactic.Linter.Lint
```

Comment:
Similarly here: the string matcher is there since one of the linters currently uses `findSubstring` or something like this.  I think that a previous version of the linter used `splitOn` and counted the pieces: that would allow to reduce the imports.

Reply:
Sure, if we can avoid increasing imports, I am happy with a different solution.  The one using `splitOn` is very low tech and yet robust, though.